components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -1069,11 +1079,28 @@ subroutine shoc_tend_e3sm( &
     enddo
    
     !  Probably need to add deepcu cloud fraction to the cloud fraction array, else would just 
-    !  be outputting the shallow convective cloud fraction 
-
+    !  be outputting the shallow convective cloud fraction
+ 
+    !  Add liq, ice, and precipitating ice fractions here. These are purely
+    !  diagnostic outputs and do not impact the rest of the code. The qi threshold for
+    !  setting ice_cloud_fraction to 1 currently set 1e-8 kg/kg but can be
```

Comment:
Putting specific values of tuning parameters in comments often leads to incorrect comments later. I'd suggest just saying "is a tunable parameter".

Reply:
Also, does the 1e-8 we use here need to be synced to the threshold used for the actual ice cloud fraction calculation used by P3? If so, we should probably make those values a single parameter set somewhere like micro_p3_utils.F90...

Reply:
And maybe also mention that the d(qi)/d(ni) slope is also a tuning parameter?